openweather/src/main/java/pages/SearchPage.java

```diff
@@ -25,16 +28,40 @@ public SearchPage(WebDriver driver) {
     }
 
     // TODO: use a common method to get row/cell text
-    public List<CityInfo> getSearchResult() {
+    public List<CityInfo>  getSearchResult() {
         List<CityInfo> cities = new ArrayList<>();
-        for (WebElement row : getTableRows()) {
-            String name = row.findElement(By.tagName("a")).getText();
-            String temperature =
-                    row.findElement(By.xpath("//*[@class='badge badge-info'")).getText();
-            cities.add(new CityInfo(name, temperature));
-        }
+            for (WebElement row : getTableRows()) {
+                String name = row.findElement(By.tagName("a")).getText();
+                String temperature =
+                        row.findElement(By.xpath("//span[@class='badge badge-info']")).getText();
+                cities.add(new CityInfo(name, temperature));
+            }
 
         return cities;
+   }
+
+    public void checkSearchResult(String searchCity) {
+        List<CityInfo> cities = getSearchResult();
+        boolean searchResult=true;
+        String msg="List of search city is returned";
+        SoftAssert softAssertion= new SoftAssert();
+
+        if (searchCity != "Not Found" && cities.size() > 0) {
+            for (CityInfo city : cities) {
+                if (!city.getCityName().contains(searchCity)) {
+                    searchResult = false;
+                    msg="Search results do not contain city";
```

Comment:
Let's consider scenarios that you want to see that info
- Running test on Jenkins, you care about test result rather than the info of running process.
- Investigating failed test, use debug mode. It's much better than log every single line
- If you really need to make sure that the data is correct, use a validation, then your test will be failed if having unexpected data 

DON'T log this info in your code.

Reply:
This info is not used for debugging only, it is used to to report a potential bugs